components/vocabularies/hpo/ui/src/main/resources/PhenoTips/HPOBranchDisplayer.xml

```diff
@@ -200,6 +200,49 @@ $xwiki.jsx.use('PhenoTips.HPOBranchDisplayer')##
     });
     $$('.hpo-branch .term-entry:not(.initialized)').invoke('addClassName', 'initialized');
   });
+});
+document.observe('xwiki:dom:loaded', function(){
+  function canCollapse(sublist){
+    return sublist.select('input').every(function(input){
+      return !input.checked
+    });
+  }
+  $$('.hpo-branch&gt;ul').forEach(function(list){
+    var listElements = list.select('li');
+    listElements.forEach(function(listElement){
+      var sublist = listElement.down('ul');
+      if (!sublist) {return;}
+      //attach click handlers
+      var expandTool = listElement.down('span.expand-tool');
+      var checkInput = listElement.down('input');
+      expandTool.observe('click', function(e) {
+        e.stop();
+        if (sublist.hasClassName('collapsed')) {
+          sublist.removeClassName('collapsed');
+          expandTool.innerHTML = "▼";
+        } else if (canCollapse(sublist)) {
+          expandTool.innerHTML = "▶";
+          sublist.addClassName('collapsed');
+        }
+      });
+      var toggleCollapse = function() {
```

Comment:
Right now its relying on a bunch of variables within the loop. I could do some closure thing but imho this is more readable.